common/lib/multer.api-document.storage.test.js

```diff
@@ -13,7 +13,11 @@ describe('Multer Document Storage Engine', function () {
   let mockReq, mockFile
 
   beforeEach(function () {
-    mockReq = {}
+    mockReq = {
+      services: {
+        document: documentService,
+      },
+    }
```

Comment:
I think you need to redefine what documentService is. Currently it's `const documentService = require('../services/document')` so it's not been replaced with a stub in this test. Something like https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/pull/1104/files#diff-7b2b1e2e9e345bf7c078fdba17df0ab89649db87375bda299c5d18bcce883b16R67 so the test isn't directly accessing the service.

Reply:
You can also to this in the mock request:

```javascript
mockReq = {
  services: {
    document: {
      create: sinon.stub(),
    },
  },
}
```

Then in your assertions check use:

```javascript
mockReq.services.document.create.resolves(mockServiceResponse)
```

and

```javascript
expect(mockReq.services.document.create).to.have.been.called
```

Reply:
Thanks! I tried `create: sinon.stub().resolves()` initially and it wasn't happy about that but without the `.resolves()` it works.